### PR TITLE
refactor: fix savestate in pre steps

### PR DESCRIPTION
**Pull request by ChristopherHX**
This now refactores the stepresult code
- pre- and post- are no longer in stepresults
- stepresults no longer has an invalid state property
- IntraActionState now stores the GITHUB_STATE

~This uncovers a design problem:~
~you can create steps with an id prefixed with pre-~

~Reduced the change to only fix a problem, tests in https://github.com/nektos/act/pull/1457 needs to be changed to enable checking this bug~

Fixes: #1459

### CODE REVIEW

File: pkg/runner/action.go [hunk 1]
```diff
@@ -371,11 +371,14 @@ func newStepContainer(ctx context.Context, step step, image string, cmd []string
 }
 
 func populateEnvsFromSavedState(env *map[string]string, step actionStep, rc *RunContext) {
-	stepResult := rc.StepResults[step.getStepModel().ID]
-	if stepResult != nil {
-		for name, value := range stepResult.State {
-			envName := fmt.Sprintf("STATE_%s", name)
-			(*env)[envName] = value
+	// design problem: you can create steps with an id prefixed with pre-
```
Comment (KnisterPeter) [inline pkg/runner/action.go:374]: I think I don't get this. Do we have a test case for that?
And why is it a problem if there is a step with a `pre-` id?

Comment (KnisterPeter) [reply pkg/runner/action.go:374]: Ouch, if that's the case it would be a critical error. Right.

Comment (KnisterPeter) [reply pkg/runner/action.go:374]: But the `step.ID` is not changed for 'pre-steps'. It's just the `step.Name` which is prefixed.
The ID should be stable and in combination with the step stage it indicates the stage of that step.

Comment (KnisterPeter) [reply pkg/runner/action.go:374]: I see, we use the `CurrentStep` in the command handler.
Wouldn't it be possible to add the `CurrentStepID` to the `RunContext` as well. That should be way more stable than a name.

Comment (KnisterPeter) [reply pkg/runner/action.go:374]: > Shouldn't we remove presteps from the steps context, called stepresults? To align to actions/runner.

Would be a good idea I think.

**Review by cplee — APPROVED** [review]

**Pull request merged**